crates/primitives/src/header.rs

```diff
@@ -311,17 +328,23 @@ impl<'a> arbitrary::Arbitrary<'a> for SealedHeader {
 impl From<Block<EthersH256>> for SealedHeader {
     fn from(block: Block<EthersH256>) -> Self {
         let header = Header {
+            parent_hash: block.parent_hash.0.into(),
             number: block.number.unwrap().as_u64(),
             gas_limit: block.gas_limit.as_u64(),
             difficulty: block.difficulty.into(),
             nonce: block.nonce.unwrap().to_low_u64_be(),
             extra_data: block.extra_data.0.into(),
             state_root: block.state_root.0.into(),
+            transactions_root: block.transactions_root.0.into(),
+            receipts_root: block.receipts_root.0.into(),
             timestamp: block.timestamp.as_u64(),
             mix_hash: block.mix_hash.unwrap().0.into(),
             beneficiary: block.author.unwrap().0.into(),
             base_fee_per_gas: block.base_fee_per_gas.map(|fee| fee.as_u64()),
-            ..Default::default()
+            ommers_hash: block.uncles_hash.0.into(),
+            gas_used: block.gas_used.as_u64(),
+            withdrawals_root: None,
+            logs_bloom: block.logs_bloom.unwrap_or_default().0.into(),
         };
```

Comment:
This is basically the implementation for `Header::From<Block>` so we should break that out and then just have `SealedHeader::From<Block>` call into it. It's cleaner plus we can create unsealed headers directly.

Reply:
I'm not following, sorry.

I don't think we have `Header::From<Block>` because Block already has the header.

this implementation is essentially just a compat impl, and should probably be phased out eventually because not actually used by reth internally.

Reply:
No worries, I was just saying that we're defining how to build a `Header` from a `Block` so we might as well implement `Header::From<Block>` and separate it from how we build a `SealedHeader` for flexibility.

Reply:
It would be a shame to lose this conversion tbh, it makes it much easier to get at the RLP encodings of block headers when using ethers-rs.

Reply:
sorry, this is all very confusing because there are multiple different Header and Block types (ethers,reth-primtive,reth-rpc). so this is hard to understand without using their fully qualified name.

Reply:
> It would be a shame to lose this implementation tbh, it makes it much easier to get at the RLP encodings of block headers when using ethers-rs.

they are fine for now but perhaps will be feature gated eventually. or additional work is done to make everything compatible

Reply:
> this is all very confusing because there are multiple different Header and Block types

Apologies, this is my first time looking into the reth code so I'm just speaking about things local to this file + ether-rs (ethers-rs `Block` going to a reth-primitives `Header`).

Reply:
gotcha,
yeh additional compat impl make sense

Reply:
fixed here https://github.com/paradigmxyz/reth/pull/1448/commits/17e66c27226d0f8d18cfbb9ede6d0c1c91dc9252